### ISSUE TITLE
Simplify the job process

### ISSUE THREAD

**Issue by knownasilya** [Strider-CD/strider#761]
Currently there are just too many separate parts, and it's hard to know what's what and where it comes into play.

I'm referring to `strider-runner-core`, `strider-simple-runner`, `lib/backchannel.js`, `job-data-monitor.js`, `job-monitor.js`, etc. (who knowns what else there is for the job workflow..).

It would be nice to simplify some of this workflow. Either by refactor or by some clear documentation.

**Comment by niallo** [comment]
Agreed. I think the biggest missing piece is clear documentation - from
there it should be easier to know what hound be refactored.

On Monday, April 20, 2015, Ilya Radchenko <notifications@github.com> wrote:

> Currently there are just too many separate parts, and it's hard to know
> what's what and where it comes into play.
>
> I'm referring to strider-runner-core, strider-simple-runner,
> lib/backchannel.js, job-data-monitor.js, job-monitor.js, etc. (who knowns
> what else there is for the job workflow..).
>
> It would be nice to simplify some of this workflow. Either by refactor or
> by some clear documentation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Strider-CD/strider/issues/761>.
>


-- 
Niall O'Higgins
W: http://niallohiggins.com
E: n@niallo.me
T: @niallohiggins

**Comment by knownasilya** [comment]
Would love the help of some of the original authors e.g. @jaredly

**Comment by jaredly** [comment]
- `strider-simple-runner`: a plugin for strider that runs jobs. [see runner plugin docs](https://github.com/Strider-CD/strider-extension-loader#runner). It is the default runner, and `strider-docker-runner` is an alternative.
- `strider-simple-runner` is largely a wrapper around the library `strider-runner-core`, which does the work of running a job, managing phases, etc. `strider-runner-core` is separate so that it could be used in other runner plugins.
- `lib/backchannel` is the interface between web clients and the backend (including plugins, runners, etc.) It payes attention to 4 events on a global event emitter, detailed [here](https://github.com/Strider-CD/strider/blob/master/lib/backchannel.js#L122)

I don't recognize `job-data-monitor` or `job-monitor`

**Comment by knownasilya** [comment]
@jaredly there is also `lib/websockets.js`..

**Comment by jaredly** [comment]
Oh right. `websockets.js` is the interface between web clients and the main emitter; `backchannel` listens for job events and orchestrates the running of jobs.

two different jobs
